lib/utils.js

```diff
@@ -143,45 +143,51 @@ var encode = function encode(str, defaultEncoder, charset, kind, format) {
     }
 
     var out = '';
-    for (var i = 0; i < string.length; ++i) {
-        var c = string.charCodeAt(i);
-
-        if (
-            c === 0x2D // -
-            || c === 0x2E // .
-            || c === 0x5F // _
-            || c === 0x7E // ~
-            || (c >= 0x30 && c <= 0x39) // 0-9
-            || (c >= 0x41 && c <= 0x5A) // a-z
-            || (c >= 0x61 && c <= 0x7A) // A-Z
-            || (format === formats.RFC1738 && (c === 0x28 || c === 0x29)) // ( )
-        ) {
-            out += string.charAt(i);
-            continue;
-        }
+    for (var j = 0; j < string.length; j += 1204) {
```

Comment:
let's store this `1204` in a constant. also, why was this number chosen?

Reply:
Great suggestion. As for why 1204 was chosen, it's based on multiple experiments found that using 1024 as the split method resulted in the fastest speed and least memory usage.

Reply:
1204 !== 1024 tho

Reply:
sry, 1024 is the correct value.  This is a foolish mistake on my part

Reply:
string.slice and array.join is undesirable. Is it due to compatibility considerations?

Reply:
Robustness - it's more methods that could be modified or deleted in the environment beforehand.

That's not a blocker here, it's just unfortunate.